### PR TITLE
Created an add_project function with improved tests and documentation

### LINKED ISSUE #49: Create project node (need to nest funding in project node)
Funding is not an element of dataset but is nested within a project node. To create a project node it is required to have a project title and project personnel and then I can add funding under project award.

### PR DESCRIPTION
**Pull request by ErinCain**
Redid the project function without all the git workflow messes! I improved the documentation and testing from the original version.

### CODE REVIEW

File: R/project-element.R [hunk 1]
```diff
@@ -0,0 +1,124 @@
+#' @title Add project element 
+#' @description This function creates a project node within the parent element that 
+#' contains all the required elements for the project section of an EML document.
+#' This function can be used in combination with add_personnel
+#' and add_funding. Add_personnel can be used to generate the project_personnel
```
Comment (thesadie) [inline R/project-element.R]: don't capitalize a function name and use code inline formatting wherever you make a reference to code in doc prose

`\code{add_personnel}`

File: R/project-element.R [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+#' @param award_information A list that includes the required funding information 
+#' for an EML document.This list must include the award title and the funderName.
+#' This list can be created by calling the add_funding function 
+#' on the funding information or by manually inputting the required information. 
+#' If the list is written manually it must be formatted as follows. 
+#'
+#' award_infomation = list(funderName = "Name", title = "Award Title")
+#'
+#' Additional information about the funding may be added to the list. See the 
+#' add_funding documentation for more information.
```
Comment (thesadie) [inline R/project-element.R]: make this a link to that function documentation `\code{\link{add_funding}}`

File: R/project-element.R [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+#'
+#' award_infomation = list(funderName = "Name", title = "Award Title")
+#'
+#' Additional information about the funding may be added to the list. See the 
+#' add_funding documentation for more information. 
+#'                         
+#' @param project_personnel A list that includes the required information on project 
+#' personnel for an EML document. It must include the first name, last name, 
+#' organization, and personnel role for this project. This list can be created 
+#' by calling the add_personnel function on the project personnel or by manually inputting the required
```
Comment (thesadie) [inline R/project-element.R]: same link to function docs

File: R/project-element.R [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    
+    if (missing(project_title) | missing(award_information) |
+        missing(project_personnel)) {
+      stop(project_error_message, call. = FALSE)
+    }
+  }
+  required_information <- c("funderName", "title", "individualName", "organization",
+                            "role")
+  
+  null_required_information <- which(c(is.null(award_information$funderName),
```
Comment (thesadie) [inline R/project-element.R:82]: you could likely combine the missing and null error handling, would be nice to revisit when i am back

**Review by thesadie — CHANGES_REQUESTED** [review]
remember to build docs

File: R/project-element.R (hunk 4 above)
Comment (ErinCain) [reply R/project-element.R:82]: I was able to combine error handling. I had much more code than I needed initially.

File: R/project-element.R (hunk 1 above)
Comment (ErinCain) [reply R/project-element.R]: Fixed.

File: R/project-element.R (hunk 2 above)
Comment (ErinCain) [reply R/project-element.R]: fixed

File: tests/testthat/test-project-elements.R [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+               "Please provide the award information.")
+  
+  expect_error(add_project(parent_element = parent_element,
+                           project_title = "my project title", 
+                           award_information = list()), 
+               "Please provide the project personnel")
+})
+
+test_that('arguments have all the nessisary required values', {
+  expect_error(add_project(parent_element = parent_element,
```
Comment (thesadie) [inline tests/testthat/test-project-elements.R:21]: After you add the default funder IDs (BOR, CDFW, CDWR), please update a few of the test to use them. unit tests are the most reliable documentation, it helps to make the "fake" arguments as realistic as possible.

**Review by thesadie — APPROVED** [review]

File: tests/testthat/test-project-elements.R (hunk 5 above)
Comment (ErinCain) [reply tests/testthat/test-project-elements.R:21]: Will do!

**Pull request merged**